query/thunk.ts

```diff
@@ -86,6 +82,9 @@ export interface ThunksApi<Ctx extends ThunkCtx> {
   ): CreateActionWithPayload<Gtx, P>;
 }
 
+let thunkInstanceCounter = 0;
+const registrationMap: Map<number, boolean> = new Map();
```

Comment:
I don't think module-level mutable variables are a great idea.  I wonder if there's a better way to track if thunks have been registered without this record-keeping.

Reply:
Perhaps it would be a more elegant solution to use a WeakMap, like:
`const registrationMap = new WeakMap<ThunksApi<any>, boolean>();`; 
Does this align better?